tokio/src/time/driver/mod.rs

```diff
@@ -335,20 +335,20 @@ impl Inner {
 
     /// Increment the number of active timeouts
     fn increment(&self) -> Result<(), Error> {
-        let mut curr = self.num.load(SeqCst);
-
         loop {
+            let curr = self.num.load(Relaxed);
             if curr == MAX_TIMEOUTS {
                 return Err(Error::at_capacity());
             }
 
-            let actual = self.num.compare_and_swap(curr, curr + 1, SeqCst);
-
-            if curr == actual {
-                return Ok(());
+            match self
+                .num
+                .compare_exchange_weak(curr, curr + 1, Release, Relaxed)
+            {
+                Err(MAX_TIMEOUTS) => return Err(Error::at_capacity()),
+                Ok(_) => return Ok(()),
+                Err(_) => continue,
```

Comment:
This match includes the observed value (with relaxed memory ordering), so we can avoid the next load by lifting the initial load out of the loop and doing this:
```rust
Err(next) => curr = next,
```
(this is how [`fetch_update` is implemented](https://doc.rust-lang.org/src/core/sync/atomic.rs.html#1774))

Note that you might want to drop the `MAX_TIMEOUTS` match at the same time.

Reply:
Ah, hey, good call. Will fix.

Reply:
Addressed in 58d62b7.